Softeq.XToolkit.DefaultAuthorization/SecuredTokenManager.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Globalization;
+using Plugin.SecureStorage;
+using Softeq.XToolkit.DefaultAuthorization.Abstract;
+
+namespace Softeq.XToolkit.DefaultAuthorization
+{
+    public abstract class SecuredTokenManagerBase : ISecuredTokenManager
+    {
+        private const string TokenKey = "SessionToken";
+        private const string TokenExpirationKey = "SessionTokenExpiresIn";
+        private const string RefreshTokenKey = "RefreshToken";
+
+        private static readonly IFormatProvider DateTimeFormatProvider = CultureInfo.InvariantCulture;
+
+        private DateTime? _tokenExpirationTime;
+
+        protected SecuredTokenManagerBase()
+        {
+            Token = CrossSecureStorage.Current.GetValue(TokenKey);
+            _tokenExpirationTime = StringToDate(CrossSecureStorage.Current.GetValue(TokenExpirationKey));
+            RefreshToken = CrossSecureStorage.Current.GetValue(RefreshTokenKey);
+        }
+
+        public string Token { get; private set; }
+
+        public bool IsTokenExpired => !_tokenExpirationTime.HasValue ||
```

Comment:
I think we should ignore case when `tokenExpirationTimespanInSeconds = 0` (unused by auth provider) and return false